adapters/dbs/mongo.js

```diff
@@ -0,0 +1,384 @@
+ * @param {number} [options.limit=this.limit] Limits the total returned documents for query will default to the this.limit value if not defined
+ * @param {number} [options.pagelength=this.pagelength] Defines the max length of each sub-set of data
+ * @param {Object|string} [options.population=this.population] The mongoose population for query will default to the this.population value if not defined
+ * @param {Object} [options.fields=this.fields] The fields that should be returned in query will default to the this.fields value if not defined
+ * @param {number} [options.skip] The number of documents to offset in query
+ * @param  {Function} cb      Callback function for search
+ */
+const _SEARCH_WITH_PAGINATION = function (options, cb) {
+	try {
+		let Model = options.model || this.model;
+		let { sort, limit, population, fields, skip, pagelength } = ['sort','limit','population','fields','skip','pagelength'].reduce((result, key) => {
+			result[key] = options[key] || this[key];
+		}, {});
+		let pages = {};
+		let total = 0;
+		let index = 0;
+		Promisie.doWhilst(() => {
+			return new Promisie((resolve, reject) => {
+				_SEARCH.call(this, { sort, limit: pagelength, fields, skip, population, model: Model }, (err, data) => {
+					if (err) reject(err);
+					else {
+						skip += data.length;
+						total += data.length;
+						pages[index++] = {
+							documents: data,
+							count: data.length
+						};
+						resolve(data.length);
+					}
+				});
+			});
+		}, current => (current === pagelength && total < limit))
+			.then(() => cb(null, pages))
+			.catch(cb);
+	}
+	catch (e) {
+		cb(e);
+	}
+};
+
+/**
+ * Convenience method for .findOne or .findById mongoose methods
+ * @param  {Object}   options Configurable options for mongo query
+ * @param {Object} [options.model=this.model] The mongoose model for query will default to the this.model value if not defined
+ * @param {string} [options.sort=this.sort] Sorting criteria for query will default to the this.sort value if not defined
+ * @param {Object|string} [options.population=this.population] The mongoose population for query will default to the this.population value if not defined
+ * @param {Object} [options.fields=this.fields] The fields that should be returned in query will default to the this.fields value if not defined
+ * @param {string} [options.docid="_id"] A field that should be queried will default to "_id"
+ * @param {Object|string|number} options.query If value is an object query will be set to the value otherwise a query will be built based on options.docid and any other value provided in options.query
+ * @param  {Function} cb      Callback function for load
+ */
+const _LOAD = function (options, cb) {
+	try {
+		let Model = options.model || this.model;
+		let { sort, population, fields, docid } = ['sort','population','fields','docid'].reduce((result, key) => {
+			result[key] = options[key] || this[key];
```

Comment:
lets have a custom util function like: https://github.com/cnkdynamics/valid-objectid/blob/master/index.js add an additional check to make sure it's exactly 24 characters before using the regex